setup.py

```diff
@@ -132,7 +132,8 @@
     "py7zr",
     "rarfile>=4.0",
     "s3fs>=2021.11.1",  # aligned with fsspec[http]>=2021.11.1
-    "tensorflow>=2.3,!=2.6.0,!=2.6.1",
+    "tensorflow>=2.3,!=2.6.0,!=2.6.1; sys_platform != 'darwin' or platform_machine != 'arm64'",
+    "tensorflow-macos; sys_platform == 'darwin' and platform_machine == 'arm64'",
```

Comment:
BTW I think `and platform_machine == 'arm64'"` is not needed, as I was able to install `tensorflow-macos` on my MacOS with an Intel chip

```
$ sysctl -a | grep Intel
machdep.cpu.vendor: GenuineIntel
machdep.cpu.brand_string: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
```

See https://github.com/huggingface/datasets-server/blob/9b346a79c5e3f51719fb2717683c4e5f86291201/services/worker/pyproject.toml#L34-L35

But maybe it's better to limit the exception cases and only use the fork for arm64.

Reply:
@severo I'm not an expert on this subject. But it appears they recommend `tensorflow-macos` also for Intel x86-64: https://developer.apple.com/metal/tensorflow-plugin/

So you think I can safely remove `and platform_machine == 'arm64'`? Or better keeping to reduce the scope?

Reply:
I really don't know, I just wanted to share my recent experience with the same problem, and use the same solution if you opt for a reduced scope.